### PR TITLE
Add very basic smoke test

### PR DESCRIPTION
**Pull request by tianon**
This runs signer+server+registry+dind to do a very simple pull+tag+push-with-signature test against the built images.

### CODE REVIEW

**Comment by tianon** [comment]
https://github.com/docker/notary-official-images/runs/6513271876?check_suite_focus=true#step:6:339 is the real money shot here:

> `Successfully signed registry.local:5000/hello-world:latest`

**Comment by tianon** [comment]
Hmm, that's not good -- added 5ca67bd to validate this, and we got https://github.com/docker/notary-official-images/runs/6513330118?check_suite_focus=true#step:6:337:

> `Successfully signed registry.local:5000/hello-world:latest`

(ie, it works fine and we don't hit what caused the revert :weary:)

**Comment by tianon** [comment]
Re-reading through https://github.com/notaryproject/notary/pull/1616, https://github.com/notaryproject/notary/issues/1602, and https://github.com/goharbor/harbor/issues/14932, I _think_ the error only manifests itself after a migration from an older version?  @mcapell @jonnystoten is that accurate?  (If so, I'm not sure there's going to be a clean way for us to smoke test for that :grimacing: :see_no_evil:)

**Comment by tianon** [comment]
Good news - https://github.com/docker/notary-official-images/commit/97764081b029591ceb385750dab21f4f8a6890cf should fail in CI now because I got `testclient.py` working in https://github.com/docker/notary-official-images/commit/4b86f3021a6d64c57f7e7e003ffae171d2da6f5f. :eyes:

**Comment by tianon** [comment]
https://github.com/docker/notary-official-images/runs/6532267077?check_suite_focus=true#step:6:1049 :partying_face: 

```
time="2022-05-20T22:00:34Z" level=debug msg="Client does not have the key to sign snapshot. Assuming that server should sign the snapshot."
...
* fatal: trust server rejected operation.
```

**Comment by tianon** [comment]
Rebased to squash, remove https://github.com/docker/notary-official-images/commit/97764081b029591ceb385750dab21f4f8a6890cf, and add the following to the commit message:

> This also runs the upstream Notary `buildscripts/testclient.py`, which catches the earlier Go issue!

**Comment by tianon** [comment]
Ok, this is ready for review (and if it's deemed good enough, merge!) :+1:

File: .test/test.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+docker run "${args[@]}" \
+	--name notary-testclient \
+	--init \
+	pypy:2.7-bullseye bash -Eeuo pipefail -xc '
+		ln -svf ../../fixtures/root-ca.crt cmd/notary/
+		pypy ./buildscripts/testclient.py \
+			--reponame registry.local:5000/testclient/testclient \
+			--server "$DOCKER_CONTENT_TRUST_SERVER"
+	'
+# TODO switch to something like "python:3.10-bullseye" once testclient.py actually supports Python 3 🙈
```
Comment (tianon) [inline .test/test.sh:121]: Opened a PR at https://github.com/notaryproject/notary/pull/1644 which I've validated with this change :eyes:

**Comment by tianon** [comment]
We're probably good to merge this, right? :innocent:

(Happy to make changes if there's things you think we could do better! :+1:)

**Comment by tianon** [comment]
Oh sorry, let me try to make that more clear. :smile:

My initial version of this PR was a simple "start up server + signer + registry, push a signed image, pull that signed image" and for some reason that simplified flow wasn't enough to trigger the Go exception.

I then managed to actually get the upstream `testclient.py` script working (which does a much more thorough test of the server/signer) and that _did_ reproduce the problem. :+1:

So it wasn't a matter of switching out the client so much as running the full upstream test suite instead of just doing a very very simplified smoke test.  I left the very basic smoke test in just because I think it's a nice simple demonstration of the bare minimum required to set this up and have it be working (and having a simplified smoke test is going to be easier to debug later if something large enough is wrong that it fails). :innocent:

**Comment by jonnystoten** [comment]
Ahhh gotcha, I understand now. Thanks for this one @tianon!

**Comment by tianon** [comment]
I'll come back and update this if/when https://github.com/notaryproject/notary/pull/1644 merges and we can update from Python 2 to Python 3 :sweat_smile:

**Pull request merged**